internal/backend/post.go

```diff
@@ -202,14 +200,17 @@ func (d *delphisBackend) NotifySubscribersOfCreatedPost(ctx context.Context, pos
 
 func (d *delphisBackend) GetPostsByDiscussionID(ctx context.Context, discussionID string) ([]*model.Post, error) {
 	iter := d.db.GetPostsByDiscussionIDIter(ctx, discussionID)
-	return d.iterToPosts(ctx, iter)
-	//return d.db.GetPostsByDiscussionID(ctx, discussionID)
+	return d.db.PostIterCollect(ctx, iter)
 }
 
 func (d *delphisBackend) GetLastPostByDiscussionID(ctx context.Context, discussionID string, minutes int) (*model.Post, error) {
 	return d.db.GetLastPostByDiscussionID(ctx, discussionID, minutes)
 }
 
+func (d *delphisBackend) GetPostsConnectionByDiscussionID(ctx context.Context, discussionID string, cursor string, limit int) (*model.PostsConnection, error) {
+	return d.db.GetPostsConnectionByDiscussionID(ctx, discussionID, cursor, limit)
```

Comment:
We may want to apply the limit check here to be strict (or at the DB level) to ensure it's never greater than X or less than Y. For instance, having a limit less than 2 will be odd for paging and having limit greater than our max (defined in the resolver) could give us some very large pages.